### PR TITLE
Allow SetCipherList to operate on SSL without modifying on SSL_CTX

### PR DESCRIPTION
**Pull request by julek-wolfssl**

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -26128,25 +26129,38 @@ int SetCipherList(WOLFSSL_CTX* ctx, Suites* suites, const char* list)
...
+#ifdef HAVE_ANON
+    byte haveAnon = 0;
+#endif
+
+    if (suites == NULL || list == NULL || (ctx == NULL && ssl == NULL)) {
         WOLFSSL_MSG("SetCipherList parameter error");
         return 0;
     }
 
+    version = ctx != NULL ? ctx->method->version : ssl->version;
```
Comment (SparkiDev) [inline src/internal.c:26144]: The only difference I can see when using ctx or ssl is that the version, privateKeySz and side are coming from ctx or ssl.
Extract the code that does the work into a new function and have 'haveAnon' be returned through a parameter.
Leave the SetCipherList API as it was and have it call the new function.
Add a new function SetCipherListEx that takes the ctx and ssl and uses the new function.

File: wolfssl/internal.h [hunk 2]
```diff
@@ -2371,8 +2371,8 @@ typedef struct TLSX TLSX;
 WOLFSSL_LOCAL int MatchSuite_ex(const WOLFSSL* ssl, Suites* peerSuites,
                                 CipherSuite* cs, TLSX* extensions);
 WOLFSSL_LOCAL int  MatchSuite(WOLFSSL* ssl, Suites* peerSuites);
-WOLFSSL_LOCAL int  SetCipherList(WOLFSSL_CTX* ctx, Suites* suites,
-                                 const char* list);
+WOLFSSL_LOCAL int  SetCipherList(const WOLFSSL_CTX* ctx, const WOLFSSL* ssl,
```
Comment (SparkiDev) [inline wolfssl/internal.h:2374]: When HAVE_ANON is defined, ctx or ssl is modified.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfssl/internal.h (hunk 2 above)
Comment (julek-wolfssl) [reply wolfssl/internal.h:2374]: I removed setting the `haveAnon` so now they are really `const`.

**Comment by julek-wolfssl** [comment]
Retest this please

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**